### PR TITLE
R pidkostelnii fe validation

### PR DESCRIPTION
**Pull request by roman454647**

### CODE REVIEW

File: client/src/components/FormValidator/index.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+              validator[rule.method] :
+              rule.method
+
+        if(validation_method(field_value, ...args, state) !== rule.validWhen) {
+          validation[rule.field] = { isInvalid: true, message: rule.message }
+          validation.isValid = false;
+        }
+      }
+    });
+    return validation;
```
Comment (InnaKutsik) [inline client/src/components/FormValidator/index.jsx]: You can rewrite it using reduce

File: client/src/components/FormValidator/index.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  validate(state) {
+    let validation = this.valid();
+    this.validations.forEach(rule => {
+      if (!validation[rule.field].isInvalid) {
+        const field_value = state[rule.field].toString();
+        const args = rule.args || [];
+        const validation_method =
+              typeof rule.method === 'string' ?
```
Comment (InnaKutsik) [inline client/src/components/FormValidator/index.jsx]: Why so hard? Typeof?

Comment (roman454647) [reply client/src/components/FormValidator/index.jsx]: removed typeof

File: client/src/components/FormValidator/index.jsx (hunk 1 above)
Comment (roman454647) [reply client/src/components/FormValidator/index.jsx]: done

File: client/src/components/Register/index.jsx [hunk 3]
```diff
@@ -1,7 +1,46 @@
...
+this.validator = new FormValidator(rules);
+
+  this.state = {
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    validation: this.validator.valid(),
+  }
+   this.submitted = false;
```
Comment (InnaKutsik) [inline client/src/components/Register/index.jsx]: ?

File: client/src/components/Register/index.jsx [hunk 4]
```diff
@@ -1,7 +1,46 @@
...
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  }
+
+  handleFormSubmit = event => {
+    event.preventDefault();
+    const validation = this.validator.validate(this.state);
+    this.setState({ validation });
+    this.submitted = true;
```
Comment (InnaKutsik) [inline client/src/components/Register/index.jsx]: Why not in state?

File: client/src/components/Register/index.jsx (hunk 3 above)
Comment (roman454647) [reply client/src/components/Register/index.jsx]: deleted

File: client/src/components/Register/index.jsx (hunk 4 above)
Comment (roman454647) [reply client/src/components/Register/index.jsx]: moved to state

File: client/src/components/Register/index.jsx (hunk 3 above)
Comment (roman454647) [reply client/src/components/Register/index.jsx]: and moved to state

**Pull request merged**